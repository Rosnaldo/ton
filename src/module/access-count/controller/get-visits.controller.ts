import { Controller, Get } from '@nestjs/common'
import { ApiTags } from '@nestjs/swagger'
import { CountApiVisitsService } from '../service/count-api'

type ReponseType = {
  status: number
  path: string
  value: number
}

@ApiTags('access-count')
@Controller('access-count')
export class AccessCountGetVisitsController {
  constructor(private readonly countApiVisitsService: CountApiVisitsService) {}
  @Get()
  async handle(): Promise<ReponseType> {
    return this.countApiVisitsService.get()
  }
}
