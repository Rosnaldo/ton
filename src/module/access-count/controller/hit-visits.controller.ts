import { Controller, Post } from '@nestjs/common'
import { ApiTags } from '@nestjs/swagger'
import { CountApiVisitsService } from '../service/count-api'

type ReponseType = {
  status: number
  path: string
  value: number
}

@ApiTags('access-count')
@Controller('access-count')
export class AccessCountHitVisitsController {
  constructor(private readonly countApiVisitsService: CountApiVisitsService) {}
  @Post('hit')
  async handle(): Promise<ReponseType> {
    return this.countApiVisitsService.hit()
  }
}
