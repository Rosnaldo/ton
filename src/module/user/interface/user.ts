import { IEntity } from 'src/util/entity'

export interface IUser extends IEntity {
  name: string
}
