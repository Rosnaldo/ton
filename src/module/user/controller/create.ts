import { Controller, Post } from '@nestjs/common'
import { ApiTags } from '@nestjs/swagger'
import { IUser } from '../interface/user'
import { UserCreateRepository } from '../repository/create.repostiory'
// import { UserFindOneRepository } from '../repository/find-one.repostiory'

@ApiTags('users')
@Controller('users')
export class UserCreateController {
  constructor(private readonly userCreateRepository: UserCreateRepository) {}

  @Post()
  async handle(): Promise<IUser> {
    return this.userCreateRepository.execute({ name: 'Dodo' })
  }
}
