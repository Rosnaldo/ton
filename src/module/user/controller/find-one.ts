import { Controller, Get, Param } from '@nestjs/common'
import { ApiTags } from '@nestjs/swagger'
import { IUser } from '../interface/user'
import { UserFindOneRepository } from '../repository/find-one.repostiory'

@ApiTags('users')
@Controller('users')
export class UserFindOneController {
  constructor(private readonly userFindOneRepository: UserFindOneRepository) {}

  @Get(':userId')
  async handle(@Param('userId') userId: string): Promise<IUser> {
    return this.userFindOneRepository.execute({ _id: userId })
  }
}
