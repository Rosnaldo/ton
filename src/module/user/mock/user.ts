import { Types } from 'mongoose'
import { IUser } from '../interface/user'

const ObjectId = Types.ObjectId

export const MakeMockUser = (): IUser => ({
  _id: ObjectId('5ce1d6bb8fef73750f97e941'),
  firstName: 'Andrey',
  lastName: 'Tsuzuki',
  email: 'andreytsuzuzki@gmail.com',
  password: 'Gtdneug32',
})
