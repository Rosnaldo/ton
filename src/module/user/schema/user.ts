import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose'
import { IUser } from '../interface/user'

@Schema()
export class User implements IUser {
  @Prop({ type: String, required: false })
  name: string
}

export type UserDocument = User & Document

export const UserSchema = SchemaFactory.createForClass(User)
