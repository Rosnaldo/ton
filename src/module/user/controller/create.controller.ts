import { Body, Controller, Post, UsePipes, ValidationPipe } from '@nestjs/common'
import { ApiTags } from '@nestjs/swagger'
import { CreateHash } from 'src/util/bcrypt'
import { CreateUserDto } from '../dto/create.dto'
import { IUser } from '../interface/user'
import { UserCreateRepository } from '../repository/create.repostiory'

@ApiTags('users')
@Controller('users')
export class UserCreateController {
  constructor(private readonly userCreateRepository: UserCreateRepository, private readonly createHash: CreateHash) {}

  @Post()
  @UsePipes(ValidationPipe)
  async handle(@Body() createUserDto: CreateUserDto): Promise<IUser> {
    createUserDto.password = await this.createHash.execute(createUserDto.password)
    return this.userCreateRepository.execute(createUserDto)
  }
}
