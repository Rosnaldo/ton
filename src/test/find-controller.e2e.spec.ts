import { FastifyAdapter, NestFastifyApplication } from '@nestjs/platform-fastify'
import { Test } from '@nestjs/testing'
import { MongooseModule } from '@nestjs/mongoose'
import { LoadSeed } from 'src/util/load-seed'
import { UserFindOneController } from 'src/module/user/controller/find-one.controller'
import { UserMongooseModule } from 'src/module/user/user.module'
import { UserFindOneRepository } from 'src/module/user/repository/find-one.repostiory'

let app: NestFastifyApplication
let replSet: any

describe('UserFindOneController', () => {
  beforeAll(async () => {
    const { uri, replSet2 } = await LoadSeed()
    replSet = replSet2
    const moduleRef = await Test.createTestingModule({
      imports: [
        MongooseModule.forRootAsync({
          useFactory: async () => ({
            uri,
            useNewUrlParser: true,
            useUnifiedTopology: true,
            useFindAndModify: false,
            useCreateIndex: true,
          }),
        }),
        UserMongooseModule,
      ],
      controllers: [UserFindOneController],
      providers: [UserFindOneRepository],
    }).compile()
    app = moduleRef.createNestApplication<NestFastifyApplication>(new FastifyAdapter())

    await app.init()
    await app.getHttpAdapter().getInstance().ready()
  })

  afterAll(async () => {
    await app.close()
    await replSet.stop()
  })

  it(`/GET user`, () => {
    return app
      .inject({
        method: 'GET',
        url: '/users/5ce5895fc312a6648e06d5f3',
      })
      .then((result) => {
        expect(result.statusCode).toEqual(200)
        expect(result.payload).toMatchSnapshot()
      })
  })
})
