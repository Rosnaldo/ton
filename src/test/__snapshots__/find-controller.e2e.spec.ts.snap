// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFindOneController /GET user 1`] = `"{\\"_id\\":\\"5ce5895fc312a6648e06d5f3\\",\\"firstName\\":\\"Andrey\\",\\"lastName\\":\\"Tsuzuki\\",\\"email\\":\\"andreytsuzuki@gmail.com\\",\\"password\\":\\"$2b$10$.fszeJjMf7VV2KByddRMJ.XPbmvnWPTyHiZ7QdfX0D/J2EFkCNQta\\"}"`;
